apps/arena/lib/arena/game_updater.ex

```diff
@@ -469,4 +481,73 @@ defmodule Arena.GameUpdater do
       Player.change_health(players_acc, player_id, 1)
     end)
   end
+
+  defp resolve_collisions(projectiles, players, obstacles, external_wall_id) do
+    Enum.reduce(projectiles, {projectiles, players}, fn {projectile_id, projectile},
+                                                        {projectiles_acc, players_acc} ->
+      # check if the projectiles is inside the walls
+      collides_with =
+        case projectile.collides_with do
+          [] -> [external_wall_id]
+          entities -> List.delete(entities, external_wall_id)
+        end
+
+      collided_entity = decide_collided_entity(projectile, collides_with, external_wall_id)
+
+      case {
```

Comment:
It would be nice to have this follow the same pattern as `decide_collided_entity` to make the code clearer

Reply:
Thanks Lotu, I created an issue for that (#247) I'll do it in another PR.